src/software/ai/hl/stp/play/kickoff_play_test.py

```diff
@@ -4,89 +4,173 @@
+    simulated_test_runner.blue_full_system_proto_unix_io.send_proto(Play, blue_play)
+    simulated_test_runner.yellow_full_system_proto_unix_io.send_proto(Play, yellow_play)
+
+    # Create world state
+    simulated_test_runner.simulator_proto_unix_io.send_proto(
+        WorldState,
+        create_world_state(
+            yellow_robot_locations=yellow_bots,
+            blue_robot_locations=blue_bots,
+            ball_location=ball_initial_pos,
+            ball_velocity=tbots_cpp.Vector(0, 0),
+        ),
+    )
+
+    # Always Validation
+    always_validation_sequence_set = [[]]
+    if is_friendly_test:
+        # Checks that either 0 or 1 robots are in centerCircle OR ball moves from center point
+        always_validation_sequence_set[0].append(
+            OrValidation(
+                [
+                    BallAlwaysMovesFromRest(
+                        position=tbots_cpp.Point(0, 0), threshold=0.01
+                    ),
+                    NumberOfRobotsAlwaysStaysInRegion(
+                        regions=[
+                            tbots_cpp.Field.createSSLDivisionBField().centerCircle()
+                        ],
+                        req_robot_cnt=0,
+                    ),
+                    NumberOfRobotsAlwaysStaysInRegion(
+                        regions=[
+                            tbots_cpp.Field.createSSLDivisionBField().centerCircle()
+                        ],
+                        req_robot_cnt=1,
+                    ),
+                ]
             )
-            blue_play.name = PlayName.KickoffFriendlyPlay
-            yellow_play.name = PlayName.KickoffEnemyPlay
-        else:
-            simulated_test_runner.gamecontroller.send_gc_command(
-                gc_command=Command.Type.KICKOFF, team=Team.YELLOW
+        )
+
+        # Checks that there are 6 friendly robots in friendlyHalf + centerCircle + friendlyGoal OR ball moves from center point
+        always_validation_sequence_set[0].append(
+            OrValidation(
+                [
+                    BallAlwaysMovesFromRest(
+                        position=tbots_cpp.Point(0, 0), threshold=0.01
+                    ),
+                    NumberOfRobotsAlwaysStaysInRegion(
+                        regions=[
+                            tbots_cpp.Field.createSSLDivisionBField().friendlyHalf(),
+                            tbots_cpp.Field.createSSLDivisionBField().friendlyGoal(),
+                            tbots_cpp.Field.createSSLDivisionBField().centerCircle(),
+                        ],
+                        req_robot_cnt=6,
+                    ),
+                ]
+            )
+        )
+
+    else:
+        # Checks that 0 robots are in centerCircle OR ball moves from center point
+        always_validation_sequence_set[0].append(
+            OrValidation(
+                [
+                    BallAlwaysMovesFromRest(
+                        position=tbots_cpp.Point(0, 0), threshold=0.01
```

Comment:
The SSL Robocup threshold is 0.05 m: https://robocup-ssl.github.io/ssl-rules/sslrules.html#_double_touch